pkg/controller/petset/pet_set.go

```diff
@@ -348,7 +348,13 @@ func (psc *PetSetController) syncPetSet(ps *apps.PetSet, pets []*api.Pod) (int,
 		case deletePet:
 			err = petManager.Delete(pet)
 		}
-		if err != nil {
+		switch err.(type) {
+		case errUnhealthyPet:
+			// We are not passing this error up, but we don't increment numPets if we encounter it,
+			// since numPets directly translates to petset.status.replicas
+		case nil:
+			continue
```

Comment:
nit: please either add continue or remove continue from both cases that we want to behave in a similar manner (case errUnhealthyPet and case nil)